File: src/app/common/purchase.ts
import { Address } from "./address";
import { Costumer } from "./costumer";
import { Order } from "./order";
import { OrderItem } from "./order-item";

export class Purchase {
    costumer: Costumer;
    shippingAddress : Address;
    billingAddress : Address;
    order : Order;
    orderItems : OrderItem[];
}

File: backEnd/spring-boot-ecommerce/src/main/java/com/wanderFerb/ecommerce/dto/Purchase.java
package com.wanderFerb.ecommerce.dto;

import com.wanderFerb.ecommerce.entity.Address;
import com.wanderFerb.ecommerce.entity.Customer;
import com.wanderFerb.ecommerce.entity.Order;
import com.wanderFerb.ecommerce.entity.OrderItem;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address shippingAddress;
    private Address billingAddress;
    private Order order;
    private Set<OrderItem> orderItems;

}
